File: reservation-service/src/main/java/ma/emsi/reservationservice/service/ReservationService.java
package ma.emsi.reservationservice.service;

import ma.emsi.reservationservice.dto.ReservationDto;

import java.util.List;

public interface ReservationService {

    ReservationDto create(ReservationDto dto);

    ReservationDto update(Long id, ReservationDto dto);

    void cancel(Long id);

    List<ReservationDto> findByUser(Long userId);
}

File: restaurant-frontend/src/services/reservationService.js
import api from './api';

const BASE_URL = '/reservation-service/api/reservations';

export const reservationService = {
    // Create a new reservation
    createReservation: async (reservationData) => {
        const response = await api.post(BASE_URL, reservationData);
        return response.data;
    },

    // Get all reservations for a user
    getUserReservations: async (userId) => {
        const response = await api.get(`${BASE_URL}/user/${userId}`);
        return response.data;
    },

    // Update a reservation
    updateReservation: async (id, reservationData) => {
        const response = await api.put(`${BASE_URL}/${id}`, reservationData);
        return response.data;
    },

    // Cancel a reservation
    cancelReservation: async (id) => {
        const response = await api.delete(`${BASE_URL}/${id}`);
        return response.data;
    }
};
